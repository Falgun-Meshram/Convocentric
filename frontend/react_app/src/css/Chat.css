.rce-citem-avatar{
    position: relative !important;
}

.online_status{
    position: absolute !important;
    left: 0;
    border-radius: 100%;
    height: 15px;
    width: 15px;
    background-color: green;
    border-color: green;
    top: 0px;
}