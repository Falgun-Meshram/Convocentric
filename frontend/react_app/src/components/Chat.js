
// import logo from '../logo.svg';

import React, { useState, useEffect } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faEllipsisVertical } from '@fortawesome/free-solid-svg-icons'
import { Col, Container, Row, Dropdown, InputGroup, DropdownButton, FormControl, Image } from 'react-bootstrap';
import { ChatList, Input, Button, MessageList, } from 'react-chat-elements';
import { useNavigate, useLocation } from 'react-router-dom';

import WebSocketInstance from "./Socket";
import axiosInstance from './axiosInstance';
import '../css/App.css';
import 'react-chat-elements/dist/main.css';
import { rest } from 'msw';

function Chat() {

  const location = useLocation();
  const date = new Date();

  const navigate = useNavigate();
  const [currUser, setCurrUser] = useState({});
  const [chatMessagesDict, setChatMessagesDict] = useState({});

  const [userList, setUserList] = useState([{
    avatar: 'https://facebook.github.io/react/img/logo.svg',
    alt: 'Reactjs',
    title: 'Facebook',
    subtitle: 'What are you doing?',
    date: new Date(),
    unread: 0,
    id: '1'
  }]);

  useEffect(() => {
    initWebsocket();
    let user = {};
    if (localStorage.getItem('user')) {
      user = JSON.parse(localStorage.getItem('user'));
    }
    setCurrUser(user);
    getAllUsers();
  }, [])

  const getAllUsers = () => {
    const options = {
      method: "GET",
      url: "/get_all_users"
    }
    let data = [];
    axiosInstance.request(options).then((res) => {
      res.data.users.map((item) =>
        data.push({
          title: item.username,
          id: String(item.id),
          avatar: 'https://facebook.github.io/react/img/logo.svg',
          alt: 'Reactjs',
          subtitle: 'What are you doing?',
          date: new Date(),
          unread: 0,
        })
      )
      setUserList(data);
    }).catch((err) => { 
        console.log(err); 
    })
  }

  const initWebsocket = () => {
    if(WebSocketInstance.readyState === 1){
      // pass
    }else{
      WebSocketInstance.connect();
    }
  }


  const initializeChat = (chatId, message) => {
    // message needs to have senderId and recieverId
    waitForSocketConnection(() => {

      WebSocketInstance.addCallbacks(
        (data) => setMessagesCallback(data), 
        (data) => addMessageCallback(data),
      )
      WebSocketInstance.fetchMessages(
        chatId,
        message
      );

    });
  }

  const newMessageSocketInit = (chatId, msg, senderId, recieverId) => {
    // message needs to have senderId and recieverId
    let message = {
      'content': msg,
      'chatId': chatId,
      'senderId': senderId,
      'recieverId': recieverId
    }

    waitForSocketConnection(() => {

      WebSocketInstance.addCallbacks(
        (data) => setMessagesCallback(data), 
        (data) => addMessageCallback(data),
      )
      WebSocketInstance.newChatMessage(
        message
      );
    });
  }

  const setMessagesCallback = (data) => {
    console.log(data);
    let temp = {...chatMessagesDict};
    temp[data.chat_id] = data.fetched_messages;
    setChatMessagesDict(temp);
  }

  const addMessageCallback = (data) => {
    console.log(data)
    let temp = chatMessagesDict.hasOwnProperty(data.chat_id)?chatMessagesDict[data.chat_id]:[];
    temp.push(data.content);
    setChatMessagesDict({...chatMessagesDict, [data.chat_id]: temp})
  }

  const waitForSocketConnection = (callback) => {
    setTimeout(function () {
      if (WebSocketInstance.state() === 1) {
        console.log('calling back');
        callback();
        return;
      } else {
        console.log("wait for connection...");
        waitForSocketConnection(callback);
      }
    }, 100);
  }


  const msg = [
    {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: date.setDate(date.getDate() - 4),
    },
    {
      position: 'left',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
    {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    }, {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
    {
      position: 'left',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date() - 10,
    },
    {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    }, {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
    {
      position: 'left',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date() - 10,
    }, {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
    {
      position: 'left',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date() - 10,
    },
    {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
    {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
    {
      position: 'left',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date() - 10,
    },
    {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    }, {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
    {
      position: 'left',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date() - 10,
    },
    {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    }, {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
    {
      position: 'left',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date() - 10,
    },
    {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    }, {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
    {
      position: 'left',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date() - 10,
    },
    {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
  ]
  const simpleMsg = [{
    position: 'right',
    type: 'text',
    text: 'Hey',
    date: date.setDate(date.getDate() - 4),
    replyButton: true,


  },
  {
    position: 'left',
    type: 'text',
    text: 'I am doing well',
    date: new Date(),
    replyButton: true,


  },]


  const [userMessage, setUserMessage] = useState("");
  const [currentGroupId, changeGroupId] = useState("");

  const handleOnChange = (e) => {
    setUserMessage(e.target.value);
  };

  const data = [
    {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
      id: '1'
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
      id: '2'
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    },
    {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    },
  ]

  const handleGroupClick = (group) => {

    // Define message object which has senderId and receiverId
    console.log(group);
    const chatId = group.chatId;
    let messages = { senderId: currUser.id, recieverId: group.id };
    initializeChat(1, messages);
    console.log(group);
  };

  const handleSubmit = (e) => {
    // currChatId, message, senderId, recieverId
    newMessageSocketInit(1, userMessage, 1, 2);
    console.log(`handlesubmit called`);
    console.log(userMessage);
  };
  const handleReply = () => {
    console.log('natural');

  }

  const getUserMessages = (groupId) => {
    console.log(`getUserMessages : ${groupId}`);
    if (groupId === "123")
      return msg;
    if (groupId === "456")
      return simpleMsg;
    else
      return [];
  };
  return (
    <Container fluid style={{ height: '100vh' }} >
      <Row style={{ height: '100vh' }} >
        <Col lg="5" style={{ height: '100vh', }}  >
          <div div style={{ height: '7vh' }} className='navBar'>
            <Image roundedCircle src='https://picsum.photos/200' style={{ 'width': '8%' }} />
            <Dropdown>
              <Dropdown.Toggle variant="success" id="dropdown-basic">
                <FontAwesomeIcon icon={faEllipsisVertical} />
              </Dropdown.Toggle>
              <Dropdown.Menu>
                <Dropdown.Item>New Group</Dropdown.Item>
                <Dropdown.Item>Logout</Dropdown.Item>
                <Dropdown.Item href='/profile' >Settings</Dropdown.Item>
              </Dropdown.Menu>
            </Dropdown>
          </div>
          <ChatList
            className="chat-list"
            onClick={handleGroupClick}
            dataSource={userList}
          />
        </Col>
        <Col lg="7" style={{ height: '100vh', background: 'aliceBlue' }} >
          <div style={{ height: '7vh' }}>
            Group Data
          </div>
          <div style={{ height: '82vh' }} className='message-list' >
            <MessageList
              className="message-list"
              lockable
              toBottomHeight="100%"
              dataSource={getUserMessages(currentGroupId)}
              onReplyClick={handleReply}
            />
            <Dropdown
              buttonProps={{
                text: 'Dropdown',
              }}
              items={[
                {
                  text: 'dolor'
                },
              ]} />
          </div>

          <div className='input'>
            <Input
              placeholder="Type here..."
              multiline={true}
              onChange={handleOnChange}
              rightButtons={
                <Button
                  onClick={handleSubmit}
                  color='white'
                  backgroundColor='black'
                  text='Send' />
              }
            />
          </div>
        </Col>
      </Row>
    </Container >
  );
}

export default Chat;
