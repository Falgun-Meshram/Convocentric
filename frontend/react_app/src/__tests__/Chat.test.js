import { render, screen } from '@testing-library/react';
import React from 'react';

import Chat from '../components/';Chat

describe("<Chat>", () => {
  it("simple render", () => {
    const { container } = render(<Chat />);
    // expect(screen.getAllByText("React")).toBeInTheDocument();
  });
});
