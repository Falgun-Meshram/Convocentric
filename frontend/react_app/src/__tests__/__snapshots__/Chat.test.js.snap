// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get all users 1`] = `
<div>
  <div
    class="container-fluid"
    style="height: 100vh;"
  >
    <div
      class="row"
      style="height: 100vh;"
    >
      <div
        class="col-lg-5"
        style="height: 100vh;"
      >
        <div
          class="navBar"
          style="height: 7vh;"
        >
          <img
            class="rounded-circle"
            src="https://picsum.photos/200"
            style="width: 8%;"
          />
          <div
            class="dropdown"
          >
            <button
              aria-expanded="false"
              class="dropdown-toggle btn btn-success"
              data-testid="dropdown"
              id="dropdown-basic"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-ellipsis-vertical "
                data-icon="ellipsis-vertical"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 128 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M64 360C94.93 360 120 385.1 120 416C120 446.9 94.93 472 64 472C33.07 472 8 446.9 8 416C8 385.1 33.07 360 64 360zM64 200C94.93 200 120 225.1 120 256C120 286.9 94.93 312 64 312C33.07 312 8 286.9 8 256C8 225.1 33.07 200 64 200zM64 152C33.07 152 8 126.9 8 96C8 65.07 33.07 40 64 40C94.93 40 120 65.07 120 96C120 126.9 94.93 152 64 152z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="rce-container-clist chat-list"
        >
          <div
            class="rce-container-citem"
          >
            <div
              class="rce-citem"
            >
              <div
                class="rce-citem-avatar"
              >
                <div
                  class="rce-avatar-container circle large"
                >
                  <img
                    alt="Reactjs"
                    class="rce-avatar"
                    src="https://facebook.github.io/react/img/logo.svg"
                  />
                </div>
              </div>
              <div
                class="rce-citem-body"
              >
                <div
                  class="rce-citem-body--top"
                >
                  <div
                    class="rce-citem-body--top-title"
                  >
                    test2
                  </div>
                  <div
                    class="rce-citem-body--top-time"
                  />
                </div>
                <div
                  class="rce-citem-body--bottom"
                >
                  <div
                    class="rce-citem-body--bottom-title"
                  >
                    What are you doing?
                  </div>
                  <div
                    class="rce-citem-body--bottom-tools"
                  />
                  <div
                    class="rce-citem-body--bottom-tools-item-hidden-hover"
                  />
                  <div
                    class="rce-citem-body--bottom-status"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="rce-container-citem"
          >
            <div
              class="rce-citem"
            >
              <div
                class="rce-citem-avatar"
              >
                <div
                  class="rce-avatar-container circle large"
                >
                  <img
                    alt="Reactjs"
                    class="rce-avatar"
                    src="https://facebook.github.io/react/img/logo.svg"
                  />
                </div>
              </div>
              <div
                class="rce-citem-body"
              >
                <div
                  class="rce-citem-body--top"
                >
                  <div
                    class="rce-citem-body--top-title"
                  >
                    test3
                  </div>
                  <div
                    class="rce-citem-body--top-time"
                  />
                </div>
                <div
                  class="rce-citem-body--bottom"
                >
                  <div
                    class="rce-citem-body--bottom-title"
                  >
                    What are you doing?
                  </div>
                  <div
                    class="rce-citem-body--bottom-tools"
                  />
                  <div
                    class="rce-citem-body--bottom-tools-item-hidden-hover"
                  />
                  <div
                    class="rce-citem-body--bottom-status"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="rce-container-citem"
          >
            <div
              class="rce-citem"
            >
              <div
                class="rce-citem-avatar"
              >
                <div
                  class="rce-avatar-container circle large"
                >
                  <img
                    alt="Reactjs"
                    class="rce-avatar"
                    src="https://facebook.github.io/react/img/logo.svg"
                  />
                </div>
              </div>
              <div
                class="rce-citem-body"
              >
                <div
                  class="rce-citem-body--top"
                >
                  <div
                    class="rce-citem-body--top-title"
                  >
                    test4
                  </div>
                  <div
                    class="rce-citem-body--top-time"
                  />
                </div>
                <div
                  class="rce-citem-body--bottom"
                >
                  <div
                    class="rce-citem-body--bottom-title"
                  >
                    What are you doing?
                  </div>
                  <div
                    class="rce-citem-body--bottom-tools"
                  />
                  <div
                    class="rce-citem-body--bottom-tools-item-hidden-hover"
                  />
                  <div
                    class="rce-citem-body--bottom-status"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="rce-container-citem"
          >
            <div
              class="rce-citem"
            >
              <div
                class="rce-citem-avatar"
              >
                <div
                  class="rce-avatar-container circle large"
                >
                  <img
                    alt="Reactjs"
                    class="rce-avatar"
                    src="https://facebook.github.io/react/img/logo.svg"
                  />
                </div>
              </div>
              <div
                class="rce-citem-body"
              >
                <div
                  class="rce-citem-body--top"
                >
                  <div
                    class="rce-citem-body--top-title"
                  >
                    test5
                  </div>
                  <div
                    class="rce-citem-body--top-time"
                  />
                </div>
                <div
                  class="rce-citem-body--bottom"
                >
                  <div
                    class="rce-citem-body--bottom-title"
                  >
                    What are you doing?
                  </div>
                  <div
                    class="rce-citem-body--bottom-tools"
                  />
                  <div
                    class="rce-citem-body--bottom-tools-item-hidden-hover"
                  />
                  <div
                    class="rce-citem-body--bottom-status"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-lg-7"
        style="height: 100vh; background: aliceBlue;"
      >
        <div
          style="height: 7vh;"
        >
          test1
           
          <button
            class="btn btn-danger btn-sm"
            style="margin: 10px;"
            type="button"
          >
            DELETE ALL
          </button>
           
          
        </div>
        <div
          class="message-list"
          id="messageDiv"
          style="height: 82vh;"
        >
          <div
            class="rce-container-mlist message-list"
          >
            <div
              class="rce-mlist"
            >
              <div
                class="rce-container-mbox"
              >
                <div
                  class="rce-mbox rce-mbox-right"
                >
                  <div
                    class="rce-mbox-body"
                  >
                    <div
                      class="rce-mbox-title rce-mbox-title--clear"
                    >
                      <span>
                        test1
                      </span>
                    </div>
                    <div
                      class="rce-mbox-text right"
                    >
                      looking
                    </div>
                    <div
                      class="rce-mbox-time non-copiable"
                      data-text="2 days ago"
                    />
                  </div>
                  <svg
                    class="rce-mbox-right-notch"
                    viewBox="0 0 20 20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0v20L20 0"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="input"
        >
          <div
            class="row"
            style="margin: 0px; padding: 0px;"
          >
            <div
              class="col-xl-10 col-lg-10 col-md-9 col-sm-9 col-9"
            >
              <input
                class="form-control"
                name="user_message"
                placeholder="Type here..."
                style="z-index: 10;"
                value=""
              />
            </div>
            <div
              class="col-xl-2 col-lg-2 col-md-3 col-sm-3 col-3"
            >
              <button
                class="rce-button"
                style="background-color: black; color: white; border-color: black;"
              >
                <span>
                  Send
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
